src/target/riscv/riscv-013.c

```diff
@@ -279,16 +281,21 @@ dm013_info_t *get_dm(struct target *target)
 	return dm;
 }
 
-static uint32_t set_hartsel(uint32_t initial, uint32_t index)
+static uint32_t set_hartsel(uint32_t initial, int index)
 {
+	initial &= ~DM_DMCONTROL_HASEL;
```

Comment:
A little favor for whoever reads the code:

```suggestion
	/* Use the provided dmcontrol value but clear HARTSEL & HASEL fields */
	initial &= ~DM_DMCONTROL_HASEL;
```